railties/CHANGELOG.md

```diff
@@ -1,3 +1,47 @@
+*   Credentials commands (e.g. `bin/rails credentials:edit`) now respect
+    `config.credentials.content_path` and `config.credentials.key_path` when set
+    in `config/application.rb`.
+
+    Before:
+
+      * `bin/rails credentials:edit` ignored `RAILS_ENV`, and would always edit
+        `config/credentials.yml.enc`.
+
+      * `bin/rails credentials:edit --environment foo` would create and edit
+        `config/credentials/foo.yml.enc`.
+
+      * If `config.credentials.content_path` or `config.credentials.key_path`
+        was set, `bin/rails credentials:edit` could not be used.  Editing
+        credentials required using `bin/rails encrypted:edit path/to/credentials
+        --key path/to/key`.
+
+    After:
+
+      * `bin/rails credentials:edit` will edit the credentials file that the app
+        would load for the current `RAILS_ENV`.
+
+      * `bin/rails credentials:edit` respects `config.credentials.content_path`
+        and `config.credentials.key_path` when set in `config/application.rb`.
+        Using `RAILS_ENV`, environment-specific paths can be set, such as:
+
+          ```ruby
+          # config/application.rb
+          module MyCoolApp
+            class Application < Rails::Application
+              config.credentials.content_path = "my_credentials/#{Rails.env}.yml.enc"
+
+              config.credentials.key_path = "path/to/production.key" if Rails.env.production?
+            end
+          end
+          ```
+
+      * `bin/rails credentials:edit --environment foo` will create
+        `config/credentials/foo.yml.enc` _if_ `config.credentials.content_path`
+        is not set for the `foo` environment.  It will then edit the credentials
+        file that the app would load for the `foo` environment.
```

Comment:
What happens if you do `bin/rails credentials:edit --environment foo` but `config.credentials.content_path` is not set to for the `foo` environment?

My guess based on this is that it will edit the default path (`config/credentials.yml.enc`), but it's not clear and probably worth clarifying.

Reply:
It will edit the credentials file that the app would load for the `foo` environment.

If `config.credentials.content_path` is not set, and `config/credentials/foo.yml.enc` exists, the app would load `config/credentials/foo.yml.enc` (the same behavior as in `main`).  So, `bin/rails credentials:edit --environment foo` would edit `config/credentials/foo.yml.enc`.

Reply:
I have tweaked the wording:

```diff
-      * `bin/rails credentials:edit --environment foo` will create
+      * `bin/rails credentials:edit --environment foo` will create and edit
         `config/credentials/foo.yml.enc` _if_ `config.credentials.content_path`
-        is not set for the `foo` environment.  It will then edit the credentials
-        file that the app would load for the `foo` environment.
+        has not been set for the `foo` environment.  Ultimately, it will edit
+        the credentials file that the app would load for the `foo` environment.
```

I have also added an "Editing..." status message to `credentials:edit` so that users can be sure of which file they are editing:

```console
$ bin/rails credentials:edit
Editing config/credentials.yml.enc...
File encrypted and saved.

$ bin/rails credentials:edit --environment production
Editing config/credentials/production.yml.enc...
File encrypted and saved.

$ RAILS_ENV=production bin/rails credentials:edit
Editing config/credentials/production.yml.enc...
File encrypted and saved.
```